mt-kahypar/datastructures/partitioned_hypergraph.h

```diff
@@ -136,6 +136,25 @@ class PartitionedHypergraph {
     freeInternalData();
   }
 
+  void resetData() {
+    _is_gain_cache_initialized = false;
+    tbb::parallel_invoke([&] {
+    }, [&] {
+      _part_ids.assign(_part_ids.size(), kInvalidPartition);
+    }, [&] {
+      _pins_in_part.data().assign(_pins_in_part.data().size(), 0);
+    }, [&] {
+      _connectivity_set.reset();
+    }, [&] {
+      for (auto& x : _part_weights) x.store(0, std::memory_order_relaxed);
+    }, [&] {
+      for (auto& x : _move_to_penalty) x.store(0, std::memory_order_relaxed);
```

Comment:
I think gain caching doesn't have to be erased, as the initialization just stores the values.